src/App.tsx

```diff
@@ -15,36 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+interface FilterParams<T> {
+  sortField: keyof T | '';
+  reversed: boolean;
+}
+
+function getPreparedGoods<T>(
+  goods: string[],
+  { sortField, reversed }: FilterParams<T>,
+): string[] {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
```

Comment:
The `sortField` state is initialized as an empty string, which might not align with the `FilterParams<T>` interface that expects `sortField` to be of type `keyof T | ''`. Consider defining a union type for `sortField` to ensure type safety, such as `const [sortField, setSortField] = useState<'alphabet' | 'length' | ''>('');`.